syntax = "proto3";

import "data_model_config.proto";
import "params_config.proto";
import "body_config.proto";

message MethodConfig {
    string path = 1;
    Props props = 2;
    ParamsConfig params = 3;
    repeated MethodResponseConfig responses = 4;
}

message MethodResponseConfig {
    int32 code = 1;
    MethodResponseType type = 4;
    Props props = 2;
    BodyConfig body = 3;
}

enum MethodResponseType {
    OK = 0;
    ERR = 1;
    INFO = 2;
}
